trin-storage/src/versioned/id_indexed_v1/store.rs

```diff
@@ -510,7 +510,11 @@ impl<TContentKey: OverlayContentKey> IdIndexedV1Store<TContentKey> {
             self.usage_stats.entry_count -= to_delete;
             self.usage_stats.total_entry_size_bytes -= deleted_content_size;
             self.usage_stats.report_metrics(&self.metrics);
-            deleted_content.extend(deleted_content_values);
+            deleted_content.extend(
```

Comment:
instead of doing it here, conversion should be done while reading data from db. Somewhere around line 481.
For example, something like this:
```rust
let key_bytes: Vec<u8> = row.get("content_key")?;
let value_bytes: Vec<u8> = row.get("content_value")?;
let value = RawContentValue::from(value_bytes);
let size: u64 = row.get("content_size")?;
TContentKey::try_from_bytes(key_bytes)
    .map(|key| (key, value, size))
    .map_err(|e| {
        rusqlite::Error::FromSqlConversionFailure(0, Type::Blob, e.into())
    })
```